### ISSUE TITLE
cannot parse type: ["string", "null"] and type:["object", "null"]

### ISSUE THREAD

**Issue by wuhangchen2279-zz** [ritchieanesco/json-schema-yup-transform#54]
With the recent merge of  https://github.com/ritchieanesco/json-schema-yup-transform/pull/42/files, yup transformer cannot parse `type: ["string", "null"]` `type:["object", "null"]`

It currently throwing below exception
TypeError: sourceVal.concat is not a function

    at prependDeep (/Users/wuh2/project/json-schema-yup-transform-master/node_modules/yup/lib/util/prependDeep.js:27:72)
    at prependDeep (/Users/wuh2/project/json-schema-yup-transform-master/node_modules/yup/lib/util/prependDeep.js:29:48)
    at ObjectSchema.concat (/Users/wuh2/project/json-schema-yup-transform-master/node_modules/yup/lib/mixed.js:202:41)
    at ObjectSchema.concat (/Users/wuh2/project/json-schema-yup-transform-master/node_modules/yup/lib/object.js:254:48)
    at Object.exports.buildProperties (/Users/wuh2/project/json-schema-yup-transform-master/src/yup/builder/index.ts:34:51)
    at Object.exports.build [as default] (/Users/wuh2/project/json-schema-yup-transform-master/src/yup/builder/index.ts:244:16)
    at Object.convertToYup [as default] (/Users/wuh2/project/json-schema-yup-transform-master/src/index.ts:15:32)
    at Object.<anonymous> (/Users/wuh2/project/json-schema-yup-transform-master/test/yup/object.test.ts:186:33)
    at Object.asyncJestTest (/Users/wuh2/project/json-schema-yup-transform-master/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:100:37)
    at /Users/wuh2/project/json-schema-yup-transform-master/node_modules/jest-jasmine2/build/queueRunner.js:45:12

This is the JSON schema input I am trying to transform
```
{
      type: "object",
      $schema: "http://json-schema.org/draft-07/schema#",
      $id: "test",
      title: "Test",
      definitions: {
        address: {
          type: "object",
          properties: {
            street_address: { type: ["string", "null"] }
          },
        }
      },
      properties: {
        mailingAddress: {
          $ref: "#/definitions/address"
        }
      }
    }
```

**Comment by ritchieanesco** [comment]
@wuhangchen2279 I've added a temporary workaround here: https://github.com/ritchieanesco/json-schema-yup-transform/commit/2f8e3ca36e253e9a7e02fde2f532bc29dc7fac24 and released in [v1.6.4](https://github.com/ritchieanesco/json-schema-yup-transform/releases/tag/v1.6.4). A proper fix would probably require some breaking changes.

**Comment by ritchieanesco** [comment]
Closing this issue. Please raise another issue if issues persist.
